### PR TITLE
IGNITE-16532 Java thin: add heartbeats

### PR DESCRIPTION
**Pull request by ptupitsyn**
Implement heartbeats in Java thin client: https://cwiki.apache.org/confluence/display/IGNITE/IEP-83+Thin+Client+Keepalive

* Add `IgniteClientConfiguration.heartbeatInterval`, default 30 seconds.
* Add `ClientConnectorConfiguration.idleTimeout`, default 0 (disabled).
* Add `idleTimeout` to handshake.
* Set effective heartbeat interval to `Math.min(heartbeatInterval, idleTimeout / 3)`.

### CODE REVIEW

File: modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java [hunk 1]
```diff
@@ -60,6 +63,9 @@
             ProtocolVersion.V3_0_0
     );
 
+    /** Minimum supported heartbeat interval. */
+    private static final long MIN_RECOMMENDED_HEARTBEAT_INTERVAL = 50;
```
Comment (isapego) [inline modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java:67]: Don't you think 50 ms is a too short for real network envirnment?

**Review by isapego — COMMENTED** [review]
Is it possible to configure client/server so that connection will be closed if no user actions are performed on it?

File: modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java (hunk 1 above)
Comment (ptupitsyn) [reply modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java:67]: @isapego agree, what would you suggest?

Comment (isapego) [reply modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java:67]: I don't know really. How about 500 or at least 200-300?

Comment (ptupitsyn) [reply modules/client/src/main/java/org/apache/ignite/internal/client/TcpClientChannel.java]: 500 sounds good, fixed.

**Review by isapego — APPROVED** [review]

**Pull request merged**